burn-tensor/src/tensor/api/base.rs

```diff
@@ -530,6 +530,177 @@ where
             .map(|v| Self::new(v))
             .collect()
     }
+
+    /// Pads the tensor according to the specified padding configuration and mode.
+    ///
+    /// Pads the last two dimensions of a Float or Int tensor with a given padding size and value, according to the specified padding mode.
+    /// Currently, only constant padding mode is implemented, which pads the tensor with a constant value.
+    ///
+    /// # Arguments
+    ///
+    /// * `pad_size` - A `PadSize` struct specifying the size of padding in each dimension.
+    /// * `pad_mode` - The padding mode, determining how the padding is applied.
+    ///
+    /// # Returns
+    ///
+    /// A new `Tensor<B, D, K>` instance with the specified padding applied.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::{Data,PadMode,PadSize,Shape,Tensor};
+    ///
+    /// fn pad_example<B: Backend>()
+    /// where
+    ///     B::FloatElem: From<f32>,
+    /// {
+    ///    let pad_const = 1.1;
+    ///    let tensor = Tensor::<B, 4>::ones_devauto(Shape::new([2, 3, 4, 5]));
+    ///
+    ///    let padded_tensor = tensor.pad(PadSize::uniform(2), PadMode::Constant(pad_const.into()));
+    /// }    
+    ///
+    ///
+    /// ```
+
+    pub fn pad(self, pad_size: PadSize, pad_mode: PadMode<K, B>) -> Tensor<B, D, K>
+    where
+        K::Elem: Element,
+        K: Numeric<B>,
+    {
+        match pad_mode {
+            PadMode::Constant(value) => {
+                let mut padded_dims: [usize; D] = self.dims();
+
+                // Update the last two dimensions with padding
+                padded_dims[D - 2] += pad_size.top + pad_size.bottom;
+                padded_dims[D - 1] += pad_size.left + pad_size.right;
+
+                // Create the ranges for the padded tensor
+                let ranges: [Range<usize>; D] = padded_dims
+                    .iter()
+                    .enumerate()
+                    .map(|(i, &dim)| {
+                        if i == D - 2 {
+                            pad_size.top..dim - pad_size.bottom
+                        } else if i == D - 1 {
+                            pad_size.left..dim - pad_size.right
+                        } else {
+                            0..dim
+                        }
+                    })
+                    .collect::<Vec<Range<usize>>>()
+                    .try_into()
+                    .unwrap();
+
+                // Create the padded tensor
+                let padded_shape = Shape::from(padded_dims);
+                let padded_tensor = Tensor::full_devauto(padded_shape, value);
```

Comment:
You can merge with `main` and add a device argument to avoid using the default device.